lib/spack/spack/package.py

```diff
@@ -83,6 +83,26 @@
 _spack_configure_argsfile = 'spack-configure-args.txt'
 
 
+def has_test_method(pkg):
+    """Determine if the package has an associated stand-alone `test` method.
+    The default, empty method provided by PackageBase is not included.
+
+    Args:
+        pkg (str): the package being checked
+
+    Returns:
+        (bool): ``True`` if the package overrides the default method; else
+            ``False``
+    """
+    if not inspect.isclass(pkg):
+        tty.die('{0}: is not a class, it is {1}'.format(pkg, type(pkg)))
+
+    return (
+        (issubclass(pkg, PackageBase) and pkg.test != PackageBase.test) or
+        (isinstance(pkg, PackageBase) and pkg.test.__func__ != PackageBase.test)
+    )
+
+
```

Comment:
The changes in this file need to be undone, since they have been introduced already in #28813

Reply:
As you can see I rebased.

Any objections to moving the `docstring`?

Reply:
@tldahlgren No objections to update the docstring. We should just remove one of the two redundant implementations.